### ISSUE TITLE
Live environment unable to authenticate

### ISSUE THREAD

**Issue by alanapplyby** [ADmad/cakephp-social-auth#98]
There seems to be an issue with this plugin:

On my staging environment and local - all works fine.

Same code from GIT cloned to live environment, I click the button my Cake4 app to auth, I am redirected to Facebook where I confirm I wish to give my app permission. The call back is made and when redirected back to my website the login screen redirect to the url - almost as if re-directed to login because of auth failed:

/users/login#_=_

As mentioned, same code on my staging environment (same serve on subdomain) works - makes no sense.

'getUserCallback' => 'getUser',

If I put a die in the first line of getUser in the Users model file it doesn't hit! I have checked server and cake logs - nothing!

I am at a loss.

**Comment by ADmad** [comment]
If you have `logErrors` config enabled then an error should be logged in case authentication with the external provider fails. 

Since it's works on your staging environment it's most likely some config or environment isssue on your live server. There's isn't much I can help you with here, you'll have to figure it yourself.

**Comment by alanapplyby** [comment]
I debugged this a little and discovered the issue. Looking at:

SocialAuthMiddleware ::__getUser

Which is looking for a profile in my social_profiles table

In my case I think I had an outdated profile in the social_profiles, the identifier field was causing a conflict from a previous initial login - I guess here:

$this->_profileModel->aliasField('identifier') => $identity->id,

Changing the value of the identifier field allowed the login.

The page I was being redirected to (login) came from the I$user check:

```
return $response->withLocation(
    Router::url($config['loginUrl'], true)
);
```

Is there a way that I can modify this to set the flash message to reflect the error, would I have to extend the middleware?

In conclusion I had forget to reset my social_profiles table when I reset my db for live launch - should anyone else encounter this issue.

Thanks.

**Comment by ADmad** [comment]
Flash messages are for end user messages not developer errors. If you were ending up with duplicate records for the same `user_id` and `provider` pair in the `social_profiles` table due to not clearing old records then setting a unique contraint for the pair could help as it would throw a db error.
